ready: quick version for GetVoidPointer

CharlesGueunet opened the pull request:
Dear Julien,

This PR contains the quick version for the `GetVoidPointer` fix.
It contains a helper function that emulate functions removed from VTK for when AOS is used:
* [x] GetVoidPointer
* [x] WritePointer ( :warning: vtkCellData changes in VTK 9, do not use `cells->GetData()` directly)
* [x] _SetVoidArray_

This PR is a step toward VTK 9 but is not enough.

Performance are not affected by these changes.

Best,
Charles

CharlesGueunet committed:
fix(GetVoidPointer): replace use by a helper

to update to new VTK that have dropped the vtkArray::GetVoidPointer
method

CharlesGueunet committed:
fix(GetVoidPointer): for current VTK first

CharlesGueunet committed:
remove more GetVoidPointer methods

CharlesGueunet committed:
chore(Misc): fix for cmake concerning python path + GetVoidPointer use

CharlesGueunet committed:
fix(PythonPath): work with both abs and rel VTK_PYTHONPATH

CharlesGueunet committed:
fix(VTK): also replace WritePointer function

CharlesGueunet committed:
chore(VTK): Add WriteVoidPointer and SetVoidArray

julien-tierny commented:
hi charles,

this one is marked "3 of 3". 
is it ready to go?

(when ready, thanks for renaming your PR from "WIP" into something like "(ready)")

cheers,
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

On Tuesday, 4 February 2020 16:38:35 CET Charles Gueunet wrote:
> Dear Julien,
> 
> This PR contains the quick version for the `GetVoidPointer` fix.
> **It is not ready yet.**
> It contains a helper function that emulate the old `GetVoidPointer` function (`ttkUtils::GetVoidPointer`) for when AOS is used.
> 
> Best,
> Charles
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/topology-tool-kit/ttk/pull/328
> 
> -- Commit Summary --
> 
>   * fix(GetVoidPointer): replace use by a helper
>   * fix(GetVoidPointer): for current VTK first
> 
> -- File Changes --
> 
>     M core/vtk/ttkAlgorithm/ttkAlgorithm.cpp (19)
>     M core/vtk/ttkAlgorithm/ttkUtils.h (18)
>     M core/vtk/ttkTriangulationAlgorithm/ttk.module (1)
>     M core/vtk/ttkTriangulationAlgorithm/ttkTriangulation.cpp (141)
> 
> -- Patch Links --
> 
> https://github.com/topology-tool-kit/ttk/pull/328.patch
> https://github.com/topology-tool-kit/ttk/pull/328.diff
> 
>

CharlesGueunet commented:
Hello Julien,

I still need some additional check to see if I have missed another removed method (hence the WIP).
The current state is consistent and stable, so it can be used if we want to merge this PR quickly.

> (when ready, thanks for renaming your PR from "WIP" into something like "(ready)")

I will do that from now on.

Charles

julien-tierny commented:
> I still need some additional check to see if I have missed another removed method (hence the WIP).
sounds good. just let me now when you're finished with them then.
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

On Friday, 7 February 2020 09:35:50 CET Charles Gueunet wrote:
> Hello Julien,
> 
> I still need some additional check to see if I have missed another removed method (hence the WIP).
> The current state is consistent and stable, so it can be used if we want to merge this PR quickly.
> 
> > (when ready, thanks for renaming your PR from "WIP" into something like "(ready)")
> 
> I will do that from now on.
> 
> Charles
> 
>

CharlesGueunet committed:
chore(GetVoidPointer): use ttkUtils + refacto

Warning, the WritePointer function is ready but can't be changed
in the code yet as the GetData() method will not be available in VTK
master (use GetConnectivity instead)

CharlesGueunet committed:
feat(GetVoidPointer): add version for vtkPoints

And port new filters

julien-tierny commented:
hi charles,

thanks for the ready status on this one.

I'm a bit confused though: it seems that you migrated some of the modules to the new GetVoidPointer/WritePointer implementation but not all.
can you clarify that please?

thanks!
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

On Tuesday, 4 February 2020 16:38:35 CET Charles Gueunet wrote:
> Dear Julien,
> 
> This PR contains the quick version for the `GetVoidPointer` fix.
> **It is not ready yet.**
> It contains a helper function that emulate the old `GetVoidPointer` function (`ttkUtils::GetVoidPointer`) for when AOS is used.
> 
> Best,
> Charles
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/topology-tool-kit/ttk/pull/328
> 
> -- Commit Summary --
> 
>   * fix(GetVoidPointer): replace use by a helper
>   * fix(GetVoidPointer): for current VTK first
> 
> -- File Changes --
> 
>     M core/vtk/ttkAlgorithm/ttkAlgorithm.cpp (19)
>     M core/vtk/ttkAlgorithm/ttkUtils.h (18)
>     M core/vtk/ttkTriangulationAlgorithm/ttk.module (1)
>     M core/vtk/ttkTriangulationAlgorithm/ttkTriangulation.cpp (141)
> 
> -- Patch Links --
> 
> https://github.com/topology-tool-kit/ttk/pull/328.patch
> https://github.com/topology-tool-kit/ttk/pull/328.diff
> 
>

CharlesGueunet commented:
> it seems that you migrated some of the modules to the new GetVoidPointer/WritePointer implementation but not all. can you clarify that please?

Indeed, I migrated some modules to check that my helper works as expected and is quite easy to use on various context. Now that I have checked that, I let the remaining modules to be migrated in a future hackaton.
There is only the WritePointer that is not really useful yet as the one in the helper should not be used with `GetData` in VTK 9. (This method will not really give a pointer to the real internal data then, so the write pointer will not write anything). I am thinking about a solution for this one but this is orthogonal to this PR.

julien-tierny commented:
> Now that I have checked that, I let the remaining modules to be migrated in a future hackaton.
ok. these hackathon people are getting more and more on their plate.
I hope you'll be able to be one of us :)

cheers,
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny

On Tuesday, 11 February 2020 08:47:02 CET Charles Gueunet wrote:
> > it seems that you migrated some of the modules to the new GetVoidPointer/WritePointer implementation but not all. can you clarify that please?
> 
> Indeed, I migrated some modules to check that my helper works as expected and is quite easy to use on various context. 
> Now that I have checked that, I let the remaining modules to be migrated in a future hackaton.
> There is only the WritePointer that is not really useful yet as the one in the helper should not be used with `GetData` in VTK 9. (This method will not really give a pointer to the real internal data then, so the write pointer will not write anything). I am thinking about a solution for this one but this is orthogonal to this PR.
> 
>
